internal/common/validation/podspec.go

```diff
@@ -54,6 +59,26 @@ func ValidatePodSpec(spec *v1.PodSpec, schedulingSpec *configuration.SchedulingC
 	return validatePorts(spec)
 }
 
+func validateTerminationGracePeriod(spec *v1.PodSpec, config *configuration.SchedulingConfig) error {
+	specHasTerminationGracePeriod := spec.TerminationGracePeriodSeconds != nil
+	configHasBounds := int64(config.MaxTerminationGracePeriod.Seconds()) > 0
```

Comment:
Similar to the previous comment, we should allow `MaxTerminationGracePeriod`, which this check prevents. If you want to make a setting optional, you need to either
- Make the config value a pointer so you can compare against nil (k8s does this), or
- Add a separate boolean flag that controls whether a setting is active (we do this, e.g., for Pulsar).

Because we bundle default config when we build Armada, I don't think it's a problem to expect the graceful termination settings to be set. So I don't think we need to make this setting optional.

Reply:
Okay that makes sense, I'll just leave them as values and assume they are sane to avoid the pointer-checking.